src/settings.js

```diff
@@ -110,8 +110,15 @@ var MEM_INIT_METHOD = 0;
 // value must be large enough for the program's requirements. If
 // assertions are on, we will assert on not exceeding this, otherwise,
 // it will fail silently.
+// When running with -s PROXY_TO_PTHREAD=1, this specifies the stack size of
+// the main browser thread (i.e. the code that is processing all the proxied
+// call requests), and not the stack size of the main application thread.
 var TOTAL_STACK = 5*1024*1024;
 
+// When running with -s PROXY_TO_PTHRERAD=1, this specifies the stack size
+// of the main application thread to use.
+var PROXIED_MAIN_TOTAL_STACK = 5*1024*1024;
```

Comment:
Seems like it would make more sense for the proxied main to use `TOTAL_STACK`.. that way the amount of stack allocated to main won't change simply by adding `PROXY_TO_PTHREAD`.

How about flipping it and making this setting control the stack needed by the prozy code (the main browser thread)?

Reply:
I guess you have  use case for independently controlling these two stack sizes?

Reply:
>  I guess you have use case for independently controlling these two stack sizes?

It is not as much about my use case, but everyone will need to control that: main browser thread will execute proxied operations from pthreads. Those proxied operations are user code, so they will need to configure how much stack that code needs.


> How about flipping it

That was my first take, but then reverted: I was not able to find a good naming for the PROXIED_MAIN_TOTAL_STACK in that scheme. MAIN_BROWSER_THREAD_STACK? but it does not always apply. PROXY_STACK? (vague). PROXIED_THREAD_STACK? (it is not a pthread, but the main thread)

Given that uses will inherently need to be aware about the stack size since it will be their code running on both sides of the fence, KISS, and TOTAL_STACK would always be the main browser thread stack, and if proxying to pthread, then PROXIED_MAIN_TOTAL_STACK specifies the stack size of that pthread.

That seems more straightforward: if one passes `-s PROXY_TO_PTHREAD=1`, then one knows "I have a proxied pthread", and knows to pass `-s PROXIED_MAIN_TOTAL_STACK=x` to set up its size.

Reply:
Fair enough.

How about having `PROXIED_MAIN_TOTAL_STACK` always default to whatever `STACK_SIZE` is?     i.e. have it default 0, which means just use the `STACK_SIZE`?

Reply:
Hmm, I am not sure if that is a big win.. it might be convenient to be able to set these in one go, although the default pthread stack size is not governed by STACK_SIZE either.. so I'd be leaning on keeping these separate?